### DISCUSSION TITLE
PDF from POST Body

### DISCUSSION THREAD

**Discussion by cdumange** [gotenberg/gotenberg#558] [Q&A]
Hey,

First & foremost, great project. PDF is most of the time a complicated subject & those kind of tools are life saving !

Now, quick question. A bit of context. I would like to use gotenberg to generate PDF based on html templates. WIth the existing endpoint, I have to write my html in a index.html file & send it to gotenberg using the _forms/chromium/convert/html_ endpoint.

Is there a reason for not having a POST endpoint taking html as a body & constructing the PDF from it ?

Keep up the good work ! :)

**Comment by gulien** [comment]
Hello @cdumange , 

Thanks for the kind words 😄 

You can actually do it on the client side, i.e., by creating an in-memory HTML content instead of reading an HTML file.

See for instance https://github.com/gotenberg/gotenberg-php#convert-an-html-document-to-pdf.

**Reply by stefangweichinger** [reply]
Let me chime in here, I am currently researching something similar.
I'd like to integrate a Gotenberg container into a workflow within [n8n](https://n8n.io).

The html should be transferred via a HTTP Request, using a file would mean I'd have to set up a shared volume between `n8n` and `gotenberg` to be able to access the html file to be converted, I assume.

We discuss this in this [ticket](https://community.n8n.io/t/sending-html-to-pdf-converter/20680/).

I wonder if in-memory HTML might be a solution here, I assume that hits memory limits with larger files?

I still try to figure out the actual way of integrating this "method", testing things.

Thanks @gulien for that amazing project, looking forward to integrate it into more and more setups here.

**Reply by stefangweichinger** [reply]
I am trying to come up with the correct HTTP request to add the HTML string to the call. Could someone provide an example, maybe?

**Reply by gulien** [reply]
Hello @stefangweichinger,

Indeed, it may hit some sort of memory limits with larger files. Another option would be to host the HTML on a web server, and use the [URL route](https://gotenberg.dev/docs/modules/chromium#url) of the Chromium module.

**Reply by cdumange** [reply]
@stefangweichinger , the first anwser by @gulien does include an example of the construction of a HTTP request to manage in memory html streaming.

The idea is to create a multipart http request. The request creation is visible here : https://github.com/gotenberg/gotenberg-php/blob/fe6cbb7d7bf4e0b254ab11ef52f85c035f5262b7/src/MultipartFormDataModule.php#L103

You can then adapt to your language :)

@gulien, thx for the anwser btw. We now have a working go client for simple usecases :)
